src/test/scala/io/delta/DeltaTableSuite.scala

```diff
@@ -37,6 +37,21 @@ class DeltaTableSuite extends QueryTest
     }
   }
 
+  test("forPath should not create a DeltaTable based on Non-Delta-Path") {
+    withTempDir { dir =>
+      testData.write.format("parquet").mode("overwrite").save(dir.getAbsolutePath)
+
+      val e1 = intercept[AnalysisException] {
+        DeltaTable.forPath(spark, dir.getAbsolutePath)
+      }.getMessage
+      assert(e1.contains(s"`${dir.getAbsolutePath}` is not a Delta table"))
+
+      val e2 = intercept[AnalysisException] {
+        DeltaTable.forPath(dir.getAbsolutePath)
+      }.getMessage
+      assert(e2.contains(s"`${dir.getAbsolutePath}` is not a Delta table"))
+    }
+  }
```

Comment:
nit: no need to test it twice. just keep the e2 test is enough.

Reply:
done.